File: python/6_kyu/data_reverse.py
# https://www.codewars.com/kata/569d488d61b812a0f7000015/train/python

def data_reverse(data):
    output, byte = [], ""
    data = "".join([str(n) for n in data])
    for i in range(len(data)):
        byte += data[i]
        if (i + 1) % 8 == 0:
            output.append(byte)
            byte = ""
    output.reverse()
    return [int(n) for n in "".join(output)]

File: javascript/6_kyu/data_reverse.js
// https://www.codewars.com/kata/569d488d61b812a0f7000015/train/javascript

function dataReverse(data) {
  const output = [];
  let byte = "";
  for (let i = 0; i < data.length; i++) {
    byte += data[i].toString();
    if ((i + 1) % 8 == 0) {
      output.push(byte);
      byte = "";
    }
  }
  output.reverse();
  return output.join("").split("").map(n => Number(n));
}
